qa/integration-tests/src/test/java/io/camunda/it/client/DecisionQueryTest.java

```diff
@@ -567,6 +569,29 @@ public void shouldRetrieveDecisionInstanceByStateAndType() {
     assertThat(result.items().size()).isEqualTo(3);
   }
 
+  @Test
+  public void shouldRetrieveDecisionInstanceByEvaluationDate() {
+    // given
+    final var allResult =
+        zeebeClient.newDecisionInstanceQuery().page(p -> p.limit(1)).send().join();
+    final var di = allResult.items().getFirst();
+
+    // when
+    final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSXX");
+    final var offsetDateTime = OffsetDateTime.parse(di.getEvaluationDate(), formatter);
+    final var result =
+        zeebeClient
+            .newDecisionInstanceQuery()
+            .filter(f -> f.evaluationDate(offsetDateTime.toString()))
```

Comment:
💬 @tmetzke, I added a QA test for the new field `evaluationDate` and noticed that our REST api uses the `OffsetDateTime` to parse dates which accepts offsets with colon only (`2024-06-05T08:29:15.027+00:00`) but the API returns results without colon (`2024-06-05T08:29:15.027+0000`).

Looks like with colon is correct:
[OpenAPI `date-time`](https://swagger.io/docs/specification/v3_0/data-models/data-types/#strings) uses the [RFC 3339](https://tools.ietf.org/html/rfc3339#section-5.6) and that  defines the offset with colon (or just `Z`):
```
time-numoffset  = ("+" / "-") time-hour ":" time-minute
```

It would be probably nicer to return dates in the same format as we accept or accept dates with offset without colon, too.

Reply:
Interesting find, I agree that it should be consistent. Can you create a follow-up issue so we can collect and find out where we return dates without offset and why?

Reply:
The `di.getEvaluationDate()` would ideally yield an `OffsetDateTime` and not a String so we can work with the date type in the controllers as mappers. The REST response model should ideally also require an OffsetDateTime object anyway.

The offset date time without colon might just be the String we receive from ElasticSearch? We should maybe accommodate for that in the transformers so the REST search layer receives a date type and doesn't have to care about the different datetime strings from different database backends.

Reply:
Yes, I think it just comes from ES right now.
Created a follow-up ticket: https://github.com/camunda/camunda/issues/24218